napi_js_props.md

```diff
@@ -0,0 +1,438 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to get and set properties on JavaScript 
+objects. Some of these types are documented under 
+[Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) of the 
+[ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Properties in JavaScript are represented as a tuple of a key and a value. 
+Fundamentally, all property keys in N-API can be represented in one of the
+following forms:
+- Named: a simple utf8 encoded string
```

Comment:
"UTF8-encoded". IINM we agreed on all caps, no dash - either way, please make sure it's consistent with our spelling of UTF8 in other places, and that it's joined to "encoded" with a hyphen.